api/tasks.go

```diff
@@ -0,0 +1,56 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/ONSdigital/dp-search-reindex-api/models"
+	"github.com/ONSdigital/dp-search-reindex-api/mongo"
+	"github.com/ONSdigital/log.go/v2/log"
+	"github.com/gorilla/mux"
+)
+
+// CreateTaskHandler returns a function that generates a new TaskName resource containing default values in its fields.
+func (api *JobStoreAPI) CreateTaskHandler(ctx context.Context) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		ctx := req.Context()
+		vars := mux.Vars(req)
+		jobID := vars["id"]
+
+		// Unmarshal task to create and validate it
+		taskToCreate := &models.TaskToCreate{}
+		if err := ReadJSONBody(ctx, req.Body, taskToCreate); err != nil {
+			log.Error(ctx, "reading request body failed", err)
+			http.Error(w, serverErrorMessage, http.StatusBadRequest)
+			return
+		}
+
+		newTask, err := api.jobStore.CreateTask(ctx, jobID, taskToCreate.TaskName, taskToCreate.NumberOfDocuments)
+		if err != nil {
+			log.Error(ctx, "creating and storing a task failed", err)
```

Comment:
Consider adding the job ID to the log data as it is useful context that will aid investigating errors.

Reply:
good idea - done